src/PHPUnit/Controller/AbstractHttpControllerTestCase.php

```diff
@@ -802,15 +802,22 @@ private function queryContentRegexAssertion($path, $pattern, $useXpath = false)
                 $path
             )));
         }
-        if (!preg_match($pattern, $result->current()->nodeValue)) {
+
+        $nodeValues = [];
+
+        foreach ($result as $node) {
+            $nodeValues[] = $node->nodeValue;
+        }
+
+        if (!preg_match($pattern, implode('', $nodeValues))) {
```

Comment:
The concern I have with this approach is that the regex might match across content boundaries.

As an example, consider this markup:

```html
<div>foo</div>
<div>bar</div>
```

and then the following assertion:

```php
$this->assertQueryContentRegex('div', '/foobar/');
```

The above, with the change you suggest here, would be a false positive.

I'd suggest doing the regex in the loop itself:

```php
$found = false;

foreach ($result as $node) {
    if (preg_match($pattern, $node->nodeValue)) {
        $found = true;
        break;
    }
}

if (! $found) {
    throw new PHPUnit_Framework_ExpectationFailedException(/* ... */);
}

$this->assertTrue($found);
```

This approach will resolve the issue you reported in #32, while ensuring we
don't get false positives due to boundary issues.